generators/client/templates/angular/src/main/webapp/app/core/login/login.service.ts.ejs

```diff
@@ -36,23 +38,17 @@ export class LoginService {
         <%_ if (websocket === 'spring-websocket') { _%>
         private trackerService: <%=jhiPrefixCapitalized%>TrackerService,
         <%_ } _%>
+        <%_ if (authenticationType === 'oauth2') { _%>
+        private location: Location,
+        <%_ } _%>
         private authServerProvider: AuthServerProvider
     ) {}
 
     <%_ if (authenticationType === 'oauth2') { _%>
     login() {
-        const port = (location.port ? ':' + location.port : '');
-        let contextPath = location.pathname;
-        if (contextPath.endsWith('accessdenied')) {
-            contextPath = contextPath.substring(0, contextPath.indexOf('accessdenied'));
-        }
-        if (!contextPath.endsWith('/')) {
-            contextPath = contextPath + '/';
-        }
-
         // If you have configured multiple OIDC providers, then, you can update this URL to /login.
         // It will show a Spring Security generated login page with links to configured OIDC providers.
-        location.href = `//${location.hostname}${port}${contextPath}oauth2/authorization/oidc`;
+        location.href = `${location.origin}${this.location.prepareExternalUrl('oauth2/authorization/oidc')}`;
```

Comment:
Are you sure this will provide the same result?

Reply:
why do you doubt? Check comments under #9769 where it's already tested